File: sdks/ts/src/sdk/models/operations/deletefilesuploadsessionsid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";

export const DeleteFilesUploadSessionsIdServerList = [
	"https://upload.box.com/api/2.0",
] as const;


export class DeleteFilesUploadSessionsIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=upload_session_id" })
  uploadSessionId: string;
}


export class DeleteFilesUploadSessionsIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  serverURL?: string;

  @SpeakeasyMetadata()
  pathParams: DeleteFilesUploadSessionsIdPathParams;
}


export class DeleteFilesUploadSessionsIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-sdk/pkg/models/operations/deletefilesuploadsessionsid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

var DeleteFilesUploadSessionsIDServerList = []string{
	"https://upload.box.com/api/2.0",
}

type DeleteFilesUploadSessionsIDPathParams struct {
	UploadSessionID string `pathParam:"style=simple,explode=false,name=upload_session_id"`
}

type DeleteFilesUploadSessionsIDRequest struct {
	Retries    *utils.RetryConfig
	ServerURL  *string
	PathParams DeleteFilesUploadSessionsIDPathParams
}

type DeleteFilesUploadSessionsIDResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
}
